arbos/tx_processor_multigas_test.go

```diff
@@ -77,3 +83,187 @@ func TestStartTxHookReturnsMultigas(t *testing.T) {
 		})
 	}
 }
+
+func TestEndTxHookMultiGasRefundNormalTx(t *testing.T) {
```

Comment:
I think this change also deserves system tests. (Both for the normal case and for retryables).